include/envoy/http/header_map.h

```diff
@@ -807,6 +807,10 @@ class RequestHeaderMap
 public:
   INLINE_REQ_STRING_HEADERS(DEFINE_INLINE_STRING_HEADER)
   INLINE_REQ_NUMERIC_HEADERS(DEFINE_INLINE_NUMERIC_HEADER)
+  virtual void setForwardingPath(absl::string_view path) PURE;
+  virtual absl::string_view getForwardingPath() PURE;
+  virtual void setFilterPath(absl::string_view path) PURE;
```

Comment:
looks like right now if a filter calls headers.Path() they'll get the original path and not the filter path

is this because this is WIP implementation? (otherwise filters should not realize that there is a difference, filters should call headers.Path() and see the filter path).

also -- without making these private methods and the config classes friends, any filter can modify and change the forwarding/filter path

Reply:
Great idea! I will make these methods private and make the caller a friend.